addons/mail/static/src/components/composer/composer.js

```diff
@@ -209,7 +209,7 @@ export class Composer extends Component {
      * @param {MouseEvent} ev
      */
     _onClickCaptureGlobal(ev) {
-        if (this.contains(ev.target)) {
+        if (this.contains(ev.target) || ev.target.closest('[data-composer-no-discard]')) {
```

Comment:
Previously we were doing some weird state management with _wasSelected in message, when checking whether to discard the current reply or not when clicking the reply button was based on the selected state at the time of the last update, because clicking on the reply button goes through this handler which would discard the message, and would cause the "isSelected" property to be false and immediately reopen the message instead of closing it.

As you've asked me to move most of the logic of actions on messages into the message model, I can't do that anymore (and I don't think it's a good solution anyway). This data attribute just lets you specify parts of the DOM that this handler should ignore. If you don't like this, I think the other option would be to have this handler do its work in the next tick and check that the event wasn't handled by replyTo.

Reply:
It's the same event, but as I said you have to do the handling in the next tick if you want to let bubbling handlers do their work first and mark the event, since the capture phase happens before the bubbling phase. I've made the change so you can take a look, it works just fine.